### PR TITLE
Document how to download files from S3

### PR DESCRIPTION
**Pull request by pableu**

### CODE REVIEW

File: docs/clients/s3.md [hunk 1]
```diff
@@ -67,3 +67,23 @@ This information is required by AWS, and cannot be guessed by AsyncAws.
...
+$fp = fopen('/path/to/file.txt', 'wb');
+fwrite($fp, $result->getBody()->getContentAsString());
+
+// Download a big file with an iterable
+$fp = fopen('/path/to/big_file.mkv', 'wb');
+$getObjectResult = $s3->GetObject([
+    'Bucket' => 'my-company-website',
+    'Key' => 'bunny.mkv',
+]);
+foreach ($result->getBody()->getChunks() as $chunk) {
```
Comment (jderusse) [inline docs/clients/s3.md:86]: I'd rather use the stream to copy the content in a file

```php
$fp = fopen('/path/to/big_file.mkv', 'wb');
stream_copy_to_stream($f->getBody()->getContentAsResource(), $s);
```

The iterrator could be use when you want to perform business logic between chunks

```php
$fp = fopen('/path/to/big_file.mkv', 'wb');
foreach ($result->getBody()->getChunks() as $chunk) {
    fwrite($fp, $chunk);
    $progress->advance();
}
```

File: docs/clients/s3.md [hunk 2]
```diff
@@ -67,3 +67,23 @@ This information is required by AWS, and cannot be guessed by AsyncAws.
 If `ContentLength` is absent, AsyncAws will read the output before sending the
 request which could have a performance impact.
 
+
+### Download files
+```
```
Comment (jderusse) [inline docs/clients/s3.md:72]: Maybe we can add a small intro telling the `GetObject` operation returns a Streamable Body that exposes 3 ways to read the content of the file.

File: docs/clients/s3.md [hunk 3]
```diff
@@ -67,3 +67,23 @@ This information is required by AWS, and cannot be guessed by AsyncAws.
...
+$getObjectResult = $s3->GetObject([
+    'Bucket' => 'my-company-website',
+    'Key' => 'humans.txt',
+]);
+$fp = fopen('/path/to/file.txt', 'wb');
+fwrite($fp, $result->getBody()->getContentAsString());
```
Comment (jderusse) [inline docs/clients/s3.md:78]: The above example is about dumping content in a file. Maybe we can leverage `getContentAsString` to perform business logic with that content

ie.
```php
$getObjectResult = $s3->GetObject([
    'Bucket' => 'my-company-website',
    'Key' => 'buildMetadata.json',
]);
$metadata = json_decode($result->getBody()->getContentAsString());
```

**Comment by Nyholm** [comment]
Thank you for this PR. It will improve the docs =)

**Comment by pableu** [comment]
Thanks for your feedback. I'll update the PR to implement the suggestions from @jderusse.

**Comment by pableu** [comment]
I updated the examples and tried to incorporate your feedback. What do you think?

**Pull request merged**